examples/sprite_batch/Batching.cpp

```diff
@@ -0,0 +1,265 @@
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Audio.hpp>
+#include <SFML/Graphics.hpp>
+
+#include <cmath>
+#include <cstdlib>
+#include <ctime>
+#include <map>
+#include <numeric>
+#include <vector>
+
+#ifdef SFML_SYSTEM_IOS
+#include <SFML/Main.hpp>
+#endif
+
+std::filesystem::path resourcesDir()
+{
+#ifdef SFML_SYSTEM_IOS
+    return "";
+#else
+    return "resources";
+#endif
+}
+
+////////////////////////////////////////////////////////////
+/// Entry point of application
+///
+/// \return Application exit code
+///
+////////////////////////////////////////////////////////////
+int main()
+{
+    std::srand(static_cast<unsigned int>(std::time(nullptr)));
+
+    // Define some constants
+    const int         gameWidth         = 800;
+    const int         gameHeight        = 600;
+    const int         maxTimesSaved     = 60;
+    const int         sameTextureChance = 85;
+    const std::size_t minSprites        = 0;
+
+
+    // Create the window of the application
+    sf::RenderWindow window(sf::VideoMode({gameWidth, gameHeight}, 32),
+                            "SFML SpriteBatch",
+                            sf::Style::Titlebar | sf::Style::Close);
+    window.setVerticalSyncEnabled(true);
+
+    std::vector<sf::Texture> textures;
+    sf::Texture              tempTexture;
+
+    int currentTex = 0;
+    while (tempTexture.loadFromFile(resourcesDir() / ("Tex" + std::to_string(currentTex) + ".png")))
+    {
+        textures.push_back(std::move(tempTexture));
+        tempTexture = sf::Texture();
```

Comment:
This is probably not going to be necessary once the move constructor for [`sf::Texture`](https://github.com/SFML/SFML/blob/master/include/SFML/Graphics/Texture.hpp) is implemented, but I'm also fine with leaving it for explicitness.

Could also be that some compilers issue a "access to moved variable" warning otherwise.